File: src/main/scala/com/pharbers/ipaas/data/driver/api/model/Operator.scala
package com.pharbers.ipaas.data.driver.api.model

/** iPaas Driver Operator 运行实体
  *
  * @author clock
  * @version 0.1
  * @since 2019/06/14 10:00
  * @note
  */
case class Operator() extends Model {
    /** Operator 包含的 Plugins
      *
      * @author clock
      * @version 0.1
      * @since 2019/06/14 11:14
      * @note
      */
    var plugin: Plugin = _

    def getPlugin: Plugin = plugin

    def setPlugin(plugin: Plugin): Unit = this.plugin = plugin
}

File: src/main/scala/com/pharbers/ipaas/data/driver/api/Annotation/Operator.java
package com.pharbers.ipaas.data.driver.api.Annotation;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.util.Map;

/**
 * 功能描述
 *
 * @author dcs
 * @version 0.0
 * @tparam T 构造泛型参数
 * @note 一些值得注意的地方
 * @since 2019/09/16 16:47
 */
@Retention(RetentionPolicy.RUNTIME)
public @interface Operator {
    boolean needPlugin() default false;
    String[] args() default {""};
    String[] source() default {"inDFName"};
    String msg() default "";
}
